src/main/java/com/hazelcast/remotecontroller/HazelcastCloudManager.java

```diff
@@ -0,0 +1,347 @@
+package com.hazelcast.remotecontroller;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.*;
+import okhttp3.Response;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import net.lingala.zip4j.ZipFile;
+import org.python.jline.internal.Log;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.*;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public class HazelcastCloudManager {
+    private static final OkHttpClient client = new OkHttpClient();
+    private String baseUrl;
+    private URI uri;
+    private final ObjectMapper mapper = new ObjectMapper();
+    private String bearerToken;
+    private static final Logger LOG = LogManager.getLogger(Main.class);
+    private static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+    private Call call;
+    private final int timeoutForClusterStateWait = 5;
+
+    public HazelcastCloudManager() {
+    }
+
+    public void loginToHazelcastCloudUsingEnvironment() throws CloudException {
+        if(System.getenv("BASE_URL") != null && System.getenv("API_KEY") != null && System.getenv("API_SECRET") != null)
+            loginToHazelcastCloud(System.getenv("BASE_URL"), System.getenv("API_KEY"), System.getenv("API_SECRET"));
+        else
+            throw new CloudException("Not all required environment variables are not set");
```

Comment:
Can you please specify which env is missing in the log message together with the values of the existing ones. Can we also do an info level log printing out the environment values (will be useful for troubleshooting)?

Reply:
I think we shouldn't log the environment values because they are API_KEY, and API_SECRET. The logged messages are uploaded as a rc log and it is public. These values shouldn't be there.

Specifying the not set environment variables is added